cdap-ui/app/cdap/components/DataPrep/helper.js

```diff
@@ -45,19 +45,21 @@ export function isCustomOption(selectedOption) {
   return selectedOption.substr(0, 6) === 'CUSTOM';
 }
 
-export function setPopoverOffset(element, footerHeight = 54) {
+export function setPopoverOffset(element, popoverLevel = 'second-level-popover') {
   let elem = element;
   let elemBounding = elem.getBoundingClientRect();
-  const FOOTER_HEIGHT = footerHeight;
 
-  let popover = document.getElementsByClassName('second-level-popover');
+  let popover = document.getElementsByClassName(popoverLevel);
   let popoverHeight = popover[0].getBoundingClientRect().height;
   let tableContainerScroll = document.getElementById('dataprep-table-id').scrollTop;
   let popoverMenuItemTop = elemBounding.top;
-  let bodyBottom = document.body.getBoundingClientRect().bottom - FOOTER_HEIGHT;
```

Comment:
@elfenheart I tried all the directives, second level and third level and this is what I am seeing. I am not sure what changed for some the `document.body.getBoundingClientRect` returns bottom that doesn't include the footer. So the removing the footer height from the bottom seemed un-necessary. 

This screenshot was taken in a browser size of 1000x600
<img width="967" alt="screen shot 2017-08-24 at 1 46 33 pm" src="https://user-images.githubusercontent.com/1452845/29687844-ac7b0a0c-88d2-11e7-8023-de03281c3539.png">


Let me know if I missed any usecase.